src/VideoReceiver/GStreamer.cc

```diff
@@ -115,10 +115,18 @@ blacklist()
     }
 
     GstPluginFeature* plugin;
-
-    if ((plugin = gst_registry_lookup_feature(reg, "bcmdec")) != nullptr) {
+    plugin = gst_registry_lookup_feature(reg, "bcmdec");
+    if (plugin != nullptr) {
         qCCritical(GStreamerLog) << "Disable bcmdec";
         gst_plugin_feature_set_rank(plugin, GST_RANK_NONE);
+        gst_object_unref(plugin);
+    }
+
+    plugin = gst_registry_lookup_feature(reg, "avdec_h264");
```

Comment:
It is better to avoid that - in case of properly configured system and stable gstreamer plugins HW decoder always have higher rank than software decoder.

If someone needs to use -bad/-ugly HW decoders with rank lower than SW decoder then yes (and their rank is lower because they are simply incomplete/unstable/use at your own risk) then he can do it but it should not be something enabled by default and affecting everyone, IMHO. Perhaps it should be activated by some environment variable or similar mechanism.

Just my 2c.